src/dmd/root/file.d

```diff
@@ -22,6 +23,376 @@ import dmd.root.filename;
 import dmd.root.rmem;
 import dmd.root.string;
 
+/**
+Encapsulated management of a memory-mapped file.
+
+The `Datum` type encodes the mapping mode: Use `ubyte` for read/write mapping
+and `const ubyte` for read-only mapping.
+*/
+struct FileMapping(Datum)
+{
+    static assert(Datum.sizeof == 1, "Not tested with other data sizes yet.");
+
+    version(Posix) enum invalidHandle = -1;
+    else version(Windows) enum invalidHandle = INVALID_HANDLE_VALUE;
+
+    // state {
```

Comment:
Interesting use of braces. I've been doing some navigation aids in cparse.d like this.

https://github.com/dlang/dmd/blob/master/src/dmd/cparse.d#L93

My text editor can do brace matching, which makes it work. You can also do things like:
```
static if (true) {
    ....
}
```